nntrainer/graph/network_graph.cpp

```diff
@@ -577,15 +633,32 @@ NetworkGraph::getGraph(const std::string &input_layer,
 
   /** copy the graph and return */
   std::vector<std::shared_ptr<Layer>> ret;
-  std::copy(layers.begin() + num_layers_remove_start,
-            layers.end() - num_layers_remove_end, std::back_inserter(ret));
+  std::transform(adj.begin() + num_layers_remove_start,
+                 adj.end() - num_layers_remove_end, std::back_inserter(ret),
+                 [](auto const &elem) { return elem.front().layer; });
```

Comment:
transform is better than std::copy?

Reply:
std::copy is better but we need a modified version of each element. Hence, std::transform is used.